### DISCUSSION TITLE
Turn on Do Not Disturb when using turn-by-turn navigation

### DISCUSSION THREAD

**Discussion by EMC-prog** [organicmaps/organicmaps#3518] [Ideas]
Hello,
I have been using Organic Maps for a while and I have thought of an idea:
As the title says, there could be an option where, while in turn by turn navigation, do not disturb mode is activated. Maybe even with the option to choose with what vehicles to use the function (for example, only when choosing car routes).

Thanks and sorry for my English, I do not speak it natively.

**Comment by biodranik** [comment]
Can you please elaborate, on what you mean by Do Not Disturb? What exactly should happen? What doesn't work right now? It is not clear to me.

**Reply by EMC-prog** [reply]
Do not disturb is a mode where notifications and calls are silenced.
This could be useful for car drivers to avoid distractions while driving.
Below this text is a concept idea.
![20221002_130850](https://user-images.githubusercontent.com/65492923/193450996-209f6647-0774-4258-9835-42dca2bb0436.jpg)

**Reply by biodranik** [reply]
Sorry, I still don't understand how is it related to Organic Maps. There are no notifications in Organic Maps at the moment. You can configure the system Do Not Disturb mode in the device if necessary. Can you please clarify what exactly is too annoying for you?

If you want to turn off voice directions announcements, you can do it from the bottom panel during the navigation.

**Reply by EMC-prog** [reply]
> You can configure the system Do Not Disturb mode in the device if necessary. 

I know, but I thought it would be a nice quality of life improvement. If it's OK, I will close this discussion.

**Reply by biodranik** [reply]
I still don't understand what exactly you want to implement or change in OM. 

And even more important, why do you want to do it, what is your issue?

Maybe you want to add a DnD button in OM that will switch the system setting? E.g. will do exactly the same what is already done by a system setting?

**Reply by EMC-prog** [reply]
> Maybe you want to add a DnD button in OM that will switch the system setting? E.g. will do exactly the same what is already done by a system setting?

Similarly, but it changes it automatically. It is not an issue, it's a function request.
I think it would be easier for some people to start driving without having to remember to put the phone in DnD.

**Reply by biodranik** [reply]
That is possible in theory, but it requires to declare in the manifest and request one additional permission. We can consider it if many users would request this feature.

https://stackoverflow.com/questions/31862753/android-how-to-turn-on-do-not-disturb-dnd-programmatically
